### DISCUSSION TITLE
Lock timeout while creating partitions using cron job

### DISCUSSION THREAD

**Discussion by asranand7** [pgpartman/pg_partman#580] [General]
When creating new partitions using cron job for a table, do we have a option to provide timeout for locks, if the DDL statement that will create. a new partition does not acquire lock within lock_timeout then fail the creation of partition. How can we do that ?

Inspired by : https://duffel.com/blog/understanding-outage-concurrency-vacuum-postgresql

**Comment by keithf4** [comment]
I would recommend setting the `lock_timeout` on a per-user or per-session basis. So either as the user running partition maintenance or for the session that is connecting to do the maintenance. Trying to build in the lock timeout to pg_partman itself would be rather tricky due to the many different options and scenarios that may require different locking levels.

I will see if there might be some way to do the session level lock for the background worker, but not sure when I'll be able to look into that further right now.a

https://www.postgresql.org/docs/16/runtime-config-client.html#GUC-LOCK-TIMEOUT
